src/plugins.ts

```diff
@@ -0,0 +1,15 @@
+import { Octokit } from "@octokit/core";
+import { OctokitOptions } from "@octokit/core/dist-types/types";
+import { HttpsProxyAgent } from "https-proxy-agent";
+
+export function autoProxyAgent(octokit: Octokit, options: OctokitOptions) {
+  const proxy = process.env["HTTPS_PROXY"] || process.env["https_proxy"];
+
+  if (!proxy) return;
+
+  const agent = new HttpsProxyAgent(proxy);
+
+  octokit.hook.before("request", (options) => {
+    options.request.agent = agent;
+  });
```

Comment:
The `options` are mutable at this point. I think instead of hooking into the request lifecycle I'd just set `options.request.agent = agent` directly.

I want to do a proper overall with the `@octokit/action` module, and turn it into something like `@octokit/app` which provides an `octokit` instance plus some app-specific APIs, instead of just being a customized `Octokit` constructor. That `@octokit/action` constructor should accept an `env` option which defaults to `process.env`, and then sets Octokit options based on that, including setting the request proxy agent. But that will be a bigger project

Reply:
Sorry for the late response @gr2m, I tried removing the hooks code like you suggested, but I ran into an issue where the fetch wrapper in core.js wasn't receiving the updated request options from the plugin here.

I might be misunderstanding how core.js and plugins work, but it looks like `@octokit/request` is configured with the request options from the default options and those passed when octokit is constructed before any plugins are run 
Request options setup: https://github.com/octokit/core.js/blob/master/src/index.ts#L108
Plugins setup: https://github.com/octokit/core.js/blob/master/src/index.ts#L165

I tried setting the agent in the `defaults` static method instead, but that led to some issues in the unit tests where the defaults always the HTTPS_PROXY environment variable value when `src/index.ts` was imported at the top of the test file, and didn't run again when changing environment variables in the test cases. This probably wouldn't be a big issue when using the module in practice, but I changed the defaults to use the function argument so that it'd read the process.env value when the Octokit class was instantiated in each test case instead of once when the module was imported.

How does that sound to you?

Reply:
> I changed the defaults to use the function argument

that is a much more elegant solution 👏🏼